File: Rest API/Python/bearer_example.py
#!/usr/bin/python

import os
from pprint import pprint

import requests
from dotenv import load_dotenv

load_dotenv()  # take environment variables from .env

# Account Info
portal = os.getenv("PORTAL")
bearer = os.getenv("BEARER")

# Request Info: Get all devices
httpVerb = 'GET'
resourcePath = '/device/devices'

# Dynamic queryParams
size = 1
offset = 0
queryParams = f'?fields=id,displayName&size={size}&offset={offset}'

# Construct URL
url = f'https://{portal}.logicmonitor.com/santaba/rest{resourcePath}{queryParams}'

# Construct headers
auth = 'bearer ' + bearer
headers = {'Content-Type': 'application/json',
           'X-version': '3', 'Authorization': auth}

# Make request
response = requests.get(url, headers=headers, verify=True, timeout=15)
response_json = response.json()

# Print response
pprint(str(response_json))

File: Rest API/Node/bearerExample.js
// Import required libraries
import axios from "axios";
import dotenv from "dotenv";

dotenv.config();

// Account Info
const portal = process.env.PORTAL;
const bearer = process.env.BEARER;

// Request Info: Get all devices
const httpVerb = "GET";
const resourcePath = "/device/devices";

// Dynamic queryParams
const size = 1;
const offset = 0;
const queryParams = `?fields=id,displayName&size=${size}&offset=${offset}`;

// Construct URL
const url = `https://${portal}.logicmonitor.com/santaba/rest${resourcePath}${queryParams}`;

// Construct headers
const headers = {
  "Content-Type": "application/json",
  "X-version": "3",
  Authorization: `Bearer ${bearer}`,
};

// Function to make the HTTP request
async function fetchDevices() {
  try {
    const response = await axios.get(url, { headers: headers, timeout: 15000 });
    console.log(response.data);
  } catch (error) {
    console.error("Error making HTTP request:", error);
  }
}

// Call the function
fetchDevices();
